src/language-js/embed.js

```diff
@@ -207,8 +207,9 @@ function uncook(cookedValue) {
   return cookedValue.replace(/([\\`]|\$\{)/g, "\\$1");
 }
 
-function escapeTemplateCharacters(doc, raw) {
-  return mapDoc(doc, currentDoc => {
+function escapeTemplateCharacters(doc, raw, mapInPlace) {
+  const mapFunc = mapInPlace ? mapDocInPlace : mapDocInPlace;
```

Comment:
This doesn't look good.

Reply:
On the other hand, `escapeTemplateCharacters` isn't exported and isn't used anywhere else, so a comment should suffice instead of this extra parameter.

Reply:
That ternary had an error in anyway that I missed - it used `mapDocInPlace` for both options. I've pushed a new commit which removes the extra parameter and just adds a comment that the original `doc` is modified.